src/analysis/convex/specific_functions.lean

```diff
@@ -71,8 +101,18 @@ begin
   refine mul_nonneg ihn _, generalize : (1 + 1) * n = k,
   cases le_or_lt m k with hmk hmk,
   { have : m ≤ k + 1, from hmk.trans (lt_add_one ↑k).le,
-    exact mul_nonneg_of_nonpos_of_nonpos (sub_nonpos.2 hmk) (sub_nonpos.2 this) },
-  { exact mul_nonneg (sub_nonneg.2 hmk.le) (sub_nonneg.2 hmk) }
+    exact mul_nonneg_of_nonpos_of_nonpos (sub_nonpos_of_le hmk) (sub_nonpos_of_le this) },
+  { exact mul_nonneg (sub_nonneg_of_le hmk.le) (sub_nonneg_of_le hmk) }
+end
+
+lemma int_prod_range_pos {m : ℤ} {n : ℕ} (hn : even n) (hm : m ∉ Ico (0 : ℤ) n) :
```

Comment:
I expected to see `sub` in the name of this lemma (and the lemma above it). These names seem pretty generic.

Reply:
Yeah... but also those lemmas can be heavily generalized so I don't know if it's worth doing anything about them for now.
Do you have a precise name in mind?